### ISSUE TITLE
Unable selecting Chinese character separately in search bar on macOS

### ISSUE THREAD

**Issue by Blueswen** [squidfunk/mkdocs-material#3975]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Chinese words can not select character separately when the search result div show up.
May be related with
1. https://github.com/squidfunk/mkdocs-material/blob/9655c3a92471f261533d48b8611a8d24dbfebb13/src/assets/javascripts/browser/keyboard/index.ts#L102
2. https://github.com/squidfunk/mkdocs-material/blob/9655c3a92471f261533d48b8611a8d24dbfebb13/src/assets/javascripts/components/search/suggest/index.ts#L91
3. https://github.com/squidfunk/mkdocs-material/blob/f4f9c21d32521db90444986d81ce801e98ac0fbf/src/assets/javascripts/components/search/query/index.ts#L128
4. [CompositionEvent](https://developer.mozilla.org/en-US/docs/Web/API/CompositionEvent)
5. https://github.com/facebook/react/issues/3926

### Expected behaviour

Able to selecting character when in composition status(between [compositionstart event](https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionstart_event) and [compositionend event](https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionend_event)) with all four arrow keys and not trigger other key event.

1. When there are some search result and text still in composition status
    1. press down arrow: able to select character with Chinese IME
    2. press up arrow: nothing happen
2. When there is a search suggestions and text still in composition status
    1. press right arrow: nothing happen, the autocomplete with suggestions show trigger after compositionend event

### Actual behaviour

1. When there are some search result and text still in composition status
    1. press down arrow: unable to selecting character and another same word shows up in input box
    2. press up arrow: another same word shows up in input box
2. When there is a search suggestions and text still in composition status
    1. press right arrow: autocomplete with suggestions been triggered and press down another same word before autocompleted shows up in input box

### Steps to reproduce

1. This bug seems only happen on macOS.
2. Need Chinese IME to reproduce this bug. Install Chinese IME ```Zhuyin - Traditional``` (in Chinese, Traditional).
3. Go to [mkdocs-material doc website](https://squidfunk.github.io/mkdocs-material/) and search.
4. Type ```5j/```, space, and ```jp6``` with ```Zhuyin - Traditional```  IME in search bar.
5. When there is a underline under text means these words are in composition status.
6. Because ```中文``` has search results and suggestions, so any of press down arrow, up arrow, or right arrow those bugs could be reproduced when ```中文``` in composition status.

### Package versions

This bug found on [mkdocs-material doc website](https://squidfunk.github.io/mkdocs-material/)

### Configuration

```yaml
This bug found on [mkdocs-material doc website](https://squidfunk.github.io/mkdocs-material/)
```

### System information

- Operating system: macOS 12.3.1
- Browser: Chrome Version 101

**Comment by Blueswen** [comment]
https://user-images.githubusercontent.com/1564148/171226576-b9f37673-8056-4799-a870-24a613879222.mov

This video shows searching with ```中文``` (with search results) in composition status and press arrow down bug. But with other words without search results like ```測試``` the composition ability is working fine.

**Comment by squidfunk** [comment]
Thanks for reporting! As discussed on Gitter, I'm unable to test Chinese search, so we're going to need to solve this together. First thing I'll try is to filter `keyup` events between `compositionstart` and `compositionend`, as suggested.

**Comment by squidfunk** [comment]
Fixed in [`95aea95`](https://github.com/squidfunk/mkdocs-material-insiders/commit/95aea95b7749eaf81a3532f697cb99f5eb35e634). I believe that this should now work as you imagined:

https://user-images.githubusercontent.com/932156/171989315-848a4058-2fc7-4985-a972-e906cd049274.mp4

Thanks for providing the very detailed example! That really helped a lot. I'm happy to improve Chinese search support.

**Comment by squidfunk** [comment]
Released as part of 8.3.1+insiders-4.17.0.
